src/CMakeLists.txt

```diff
@@ -389,6 +390,25 @@ include_directories(SYSTEM ${Exiv2_INCLUDE_DIRS})
 list(APPEND LIBS ${Exiv2_LIBRARIES})
 add_definitions(${Exiv2_DEFINITIONS})
 
+# Optionally enable ISOBMFF support in Exiv2
+if (USE_ISOBMFF)
+  if(Exiv2_VERSION VERSION_GREATER_EQUAL "0.27.4")
+    # exiv2 >= 0.27.4 provides optional support for ISOBMFF files.
+    # This must be manually enabled during exiv2 build and can be checked with
+    # the EXV_ENABLE_BMFF symbol.
+    check_cxx_symbol_exists(EXV_ENABLE_BMFF "exiv2/exiv2.hpp" HAVE_EXV_ENABLE_BMFF)
+    if(HAVE_EXV_ENABLE_BMFF)
+      add_definitions(-DHAVE_LIBEXIV2_WITH_ISOBMFF=1)
+      set(DT_SUPPORTED_EXTENSIONS ${DT_SUPPORTED_EXTENSIONS} cr3 heif CACHE INTERNAL "")
```

Comment:
Why not 'heic' as well?

Btw, even though we can read metadata from heif/heic now, we still can't decode the images since dt is not leveraging libheif AFAIK.

Reply:
Hm... I'd need to check but ISOBMFF files enabled with introduction of this should be: CR3, HEIF, HEIC & AVIF

Reply:
`avif` is already added to the list of supported extensions if we detect `libavif`, and we can actually decode those images...

So it's a bit of a hodgepodge now, what does "supported extension" mean? You can decode the image (CR3 is still missing in rawspeed, HEIF/HEIC as mentioned), you can read the metadata, any level of support...?

Reply:
IMHO, this line should be removed completely, and supported extensions added in other places once the required decoding libraries are available like it is the case for all other extensions now.

Otherwise, great approach, like it better than directly checking for EXV_ENABLE_BMFF in source code.

Reply:
And another thought: maybe it makes more sense to make sure and _remove_ `cr3` from the list if BMFF support is not detected instead?

I would still leave heif/heic, avif and jxl extensions up to library availability, as metadata is not so crucial there, they are not raw files...

Reply:
I was not aware that HEIF can also have HEIC file extension. I mainly focused for the CR3 thing. After some research, the HEIF files from Canon have the .HIF extension. So we should add all variants (and I expect there are more than this three).

I've not added avif because this is already added when libavif is found.

I would like to keep *cr3* there (as ISOBMFF exiv2 support is the only requirement if rawspeed support is done) and strip out the heif/heic/hif part and move to libheif detection.
While scrolling through the code I figured out it would be easy to implement HEIF support via libheif. I'm working on a PR for this.